### PR TITLE
CB-3365 doc: add jsdoc to resources

### PR DESCRIPTION
**Pull request by Wroud**

### CODE REVIEW

**Review by devnaumov — APPROVED** [review]

File: webapp/packages/core-sdk/src/Resource/CachedResource.ts [hunk 1]
```diff
@@ -804,6 +880,13 @@ export abstract class CachedResource<
     return key;
   }
 
+  /**
+   * Check if key is valid. Can be override to provide custom validation.
```
Comment (tfokina) [inline webapp/packages/core-sdk/src/Resource/CachedResource.ts]: Can be overridden

File: webapp/packages/core-sdk/src/Resource/CachedResource.ts [hunk 2]
```diff
@@ -815,6 +898,9 @@ export abstract class CachedResource<
     return this.validateKey(param);
   }
 
+  /**
+   * Check if key is valid. Can be override to provide custom validation.
```
Comment (tfokina) [inline webapp/packages/core-sdk/src/Resource/CachedResource.ts]: Can be overridden

File: webapp/packages/core-sdk/src/Resource/CachedResource.ts [hunk 3]
```diff
@@ -62,6 +63,9 @@ export type CachedResourceMetadata<TResource> = TResource extends CachedResource
 
 export const CachedResourceParamKey = resourceKeyAlias('@cached-resource/param-default');
 
+/**
+ * CachedResource is a base class for all resources. It used to load, cache and manage data from external sources.
```
Comment (tfokina) [inline webapp/packages/core-sdk/src/Resource/CachedResource.ts]: It is used to load

**Review by tfokina — APPROVED** [review]

File: webapp/packages/core-sdk/src/Resource/CachedResource.ts (hunk 3 above)
Comment (Wroud) [reply webapp/packages/core-sdk/src/Resource/CachedResource.ts]: fixed

File: webapp/packages/core-sdk/src/Resource/CachedResource.ts (hunk 2 above)
Comment (Wroud) [reply webapp/packages/core-sdk/src/Resource/CachedResource.ts]: fixed

File: webapp/packages/core-sdk/src/Resource/CachedResource.ts (hunk 1 above)
Comment (Wroud) [reply webapp/packages/core-sdk/src/Resource/CachedResource.ts]: fixed

**Pull request merged**